python/paddle/v2/framework/initializer.py

```diff
@@ -20,6 +24,41 @@ def __call__(self, param, block):
         """
         raise NotImplementedError()
 
+    def _compute_fans(self, var):
+        """Compute the fan_in and the fan_out for layers
+
+        This method computes the fan_in and the fan_out
+        for neural network layers, if not specified. It is
+        not possible to perfectly estimate fan_in and fan_out.
+        This method will estimate it correctly for matrix multiply and
+        convolutions.
+
+        Args:
+            var: variable for which fan_in and fan_out have to be computed
+
+        Returns:
+            tuple of two integers (fan_in, fan_out)
+        """
+        shape = var.shape
+        if not shape or len(shape) == 0:
+            fan_in = fan_out = 1
+        elif len(shape) == 1:
+            fan_in = fan_out = shape[0]
+        elif len(shape) == 2:
+            # This is the case for simple matrix multiply
+            fan_in = shape[0]
+            fan_out = shape[1]
+        else:
+            # Assume this to be a convolutional kernel
+            # In Paddlepaddle, the shape of the kernel is like:
```

Comment:
s/Paddlepaddle/PaddlePaddle/g

Reply:
Fixed in 4e1fa1b305a9d3662950eb69b5cbb1c28bc166f8